sys/kern/link_elf.c

```diff
@@ -81,6 +81,9 @@ typedef struct elf_file {
 	struct linker_file lf __subobject_member_used_for_c_inheritance; /* Common fields */
 	int		preloaded;	/* Was file pre-loaded */
 	caddr_t		address;	/* Relocation address */
+#ifdef __CHERI_PURE_CAPABILITY__
+	caddr_t		mapbase;	/* Capability for code pointers */
```

Comment:
Used for all pointers, not just code ones?

Reply:
Oh, yes.  I started off with just a pcc_cap but refined it over several iterations.